agent.go

```diff
@@ -133,6 +133,17 @@ type agent interface {
 	// to handle all other Agent interface methods.
 	init(pod *Pod, config interface{}) error
 
+	// vmURL returns the agent URL exposed by the hypervisor. This URL is
+	// directly connected to the agent running inside the virtual machine.
+	vmURL() (string, error)
+
+	// proxyURL sets the URL virtcontainers should connect in order to
+	// communicate with the agent. This URL can be either the proxy URL
+	// if it actually needs to go through a proxy, or it can be the VM
+	// URL in case no proxy is needed. Both ways, this has to be
+	// transparent for the agent implementation.
+	proxyURL(url string) error
```

Comment:
This should be `setProxyURL()` to be idiomatic go I think.

Reply:
Oh yeah you're right, otherwise this looks like a getter.

Reply:
Done